schema/me/followed_artists_artworks_group.js

```diff
@@ -0,0 +1,84 @@
+import { pageable } from "relay-cursor-paging"
+import { connectionDefinitions, connectionFromArraySlice } from "graphql-relay"
+import Artwork from "schema/artwork"
+import ArtworkSorts from "schema/sorts/artwork_sorts"
+import Image from "schema/image"
+import { GraphQLList, GraphQLObjectType, GraphQLString } from "graphql"
+import { omit, groupBy, map } from "lodash"
+import { parseRelayOptions } from "lib/helpers"
+import { GlobalIDField, NodeInterface } from "schema/object_identification"
+
+const FollowedArtistsArtworksGroupType = new GraphQLObjectType({
+  name: "FollowedArtistsArtworksGroup",
+  interfaces: [NodeInterface],
+  isTypeOf: obj => obj._type === "FollowedArtistsArtworksGroup",
+  fields: () => ({
+    __id: GlobalIDField,
+    artworks: {
+      type: new GraphQLList(Artwork.type),
+      description: "List of artworks in this group.",
+    },
+    artists: {
+      type: GraphQLString,
+    },
+    summary: {
+      type: GraphQLString,
+    },
+    image: {
+      type: Image.type,
+      resolve: ({ artworks }) => {
+        return artworks.length > 0 && artworks[0].artists.length > 0 && Image.resolve(artworks[0].artists[0])
+      },
+    },
+  }),
+})
+
+const FollowedArtistsArtworksGroup = {
+  type: connectionDefinitions({ nodeType: FollowedArtistsArtworksGroupType }).connectionType,
+  description: "A list of published artworks by followed artists (grouped by date and artists).",
+  args: pageable({
+    sort: ArtworkSorts,
+  }),
+  resolve: (root, options, request, { rootValue: { followedArtistsArtworksLoader } }) => {
+    if (!followedArtistsArtworksLoader) return null
+
+    // Convert Relay-style pagination to the supported page/size style for the backend.
+    const gravityOptions = parseRelayOptions(options)
+    gravityOptions.total_count = true
+
+    return followedArtistsArtworksLoader(omit(gravityOptions, "offset")).then(({ body, headers }) => {
+      const connection = connectionFromArraySlice(body, options, {
+        arrayLength: headers["x-total-count"],
+        sliceStart: gravityOptions.offset,
+      })
```

Comment:
Is it not possible to do the grouping _prior_ to building the connection and having to muck with the data?

Reply:
That's a good idea, I'll experiment. That may clean up some of the mucking.

Reply:
So, I'm sort of inclined to leave it as-is, the data mucking _is_ definitely lighter with not making the connection first. But not really super significantly....mostly just have to avoid copying thru the `cursor` value, and you avoid have to do `.node.` in a few places.

However, since I'm creating an array that will have less than the returned elements (and varies based on the data returned), but I only have the `x-total-count` and `offset` values to pass in to the connection helper, I've noticed when making this change that scrolling thru in Emission now reveals occasional repeated groups.

So, I'd prefer to make the connection immediately from the data and then munge the data a bit, to ensure correct pagination/scrolling.

Reply:
👍